### PR TITLE
Feature akhil order

### PR DESCRIPTION
**Pull request by Akhil1857**
POST Endpoint Created

### CODE REVIEW

**Comment by Bhavya-verma-knoldus** [comment]
When will you be making the requested changes?

File: scripts/order.json [hunk 1]
```diff
@@ -24,6 +24,19 @@
...
+      "fields": [
+        { "name": "merchant_id", "type": "string" },
+        { "name": "total", "type": "decimal" }
+      ]
+    },
+    "error": {
+      "fields":[
+        { "name": "code", "type": "string" },
+        { "name": "message", "type": "string" },
+        { "name": "details", "type": "string" }
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: no need for `details`. All the information can be stored in `message`. Also, let's make message a sequence of string.

File: scripts/order.json [hunk 2]
```diff
@@ -24,6 +24,19 @@
         { "name": "submitted_at", "type": "date-time-iso8601" },
         { "name": "total", "type": "decimal" }
       ]
+    },
+    "orderform": {
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: Please follow the convention - this should be `order_form` instead.

File: scripts/order.json [hunk 3]
```diff
@@ -39,8 +52,44 @@
             "401": { "type": "unit" },
             "404": { "type": "unit" }
           }
+        },
+        {
+        "method": "POST",
+          "body": {
+          	"type": "order"
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: the body will be of type `order_form`

File: scripts/order.json [hunk 4]
```diff
@@ -39,8 +52,44 @@
...
+          },
+          "responses": {
+            "200": { "type": "orderform" },
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: `order_form` on this line and the next line too.

File: scripts/order.json [hunk 5]
```diff
@@ -39,8 +52,44 @@
...
+            "201": { "type": "orderform" },
+            "404": { "type": "unit" },
+            "422": { "type": "[error]" }
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: this should not be a sequence of error.

File: scripts/order.json [hunk 6]
```diff
@@ -39,8 +52,44 @@
...
+          }
+        },
+        {
+        "method": "DELETE",
+        "body":{
+        	"type": "unit"
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: you can skip using the entire `body` field.

File: scripts/order.json [hunk 7]
```diff
@@ -39,8 +52,44 @@
...
+        },
+        "responses":{
+             "200": { "type": "order" },
+             "401": { "type": "unit" },
+             "404": { "type": "unit" },
+             "422": { "type": "[error]" }
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: Ditto: https://github.com/Bhavya-verma-knoldus/api/pull/2/files#r1361954211

File: scripts/order.json [hunk 8]
```diff
@@ -39,8 +52,44 @@
...
+        {
+        "method": "PUT",
+        "body": {
+          	"type": "orderform"
+          },
+          "responses": {
+            "200": { "type": "order" },
+            "201": { "type": "order" },
+            "404": { "type": "unit" },
+            "422": { "type": "[error]" }
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: Ditto: https://github.com/Bhavya-verma-knoldus/api/pull/2/files#r1361954211

**Comment by Bhavya-verma-knoldus** [comment]
The indentation seems to be a bit off at some places. Kindly take care of that.

File: scripts/order.json [hunk 9]
```diff
@@ -24,6 +24,18 @@
...
+    },
+    "order_form": {
+      "fields": [
+        { "name": "merchant_id", "type": "string" },
+        { "name": "total", "type": "decimal" }
+      ]
+    },
+    "error": {
+      "fields":[
+        { "name": "code", "type": "[string]" },
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: code will be a string and the message will be a sequence of string.

File: scripts/order.json [hunk 10]
```diff
@@ -39,8 +51,41 @@
...
+            "200": { "type": "order_form" },
+            "201": { "type": "order_form" },
+            "404": { "type": "unit" },
+            "422": { "type": "error" }
+          }
+        },
+        {
+        "method": "DELETE",
+        "responses":{
+             "200": { "type": "order" },
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: We do not return an object when we delete it. So, this will be a unit.

File: scripts/order.json (hunk 9 above)
Comment (Akhil1857) [reply scripts/order.json]: Sure..will change it

File: scripts/order.json (hunk 10 above)
Comment (Akhil1857) [reply scripts/order.json]: Okay

File: scripts/order.json [hunk 11]
```diff
@@ -39,8 +51,41 @@
...
           }
+        },
+        {
+        "method": "POST",
+          "body": {
+          	"type": "order_form"
+          },
+          "responses": {
+            "200": { "type": "order_form" },
+            "201": { "type": "order_form" },
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: Let's not use 201 here and use 401 instead with unit type

File: scripts/order.json [hunk 12]
```diff
@@ -39,8 +51,41 @@
...
+        }
+        },
+        {
+        "method": "PUT",
+        "body": {
+          	"type": "order_form"
+          },
+          "responses": {
+            "200": { "type": "order" },
+            "201": { "type": "order" },
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: Ditto: https://github.com/Bhavya-verma-knoldus/api/pull/2/files#r1363747814

File: scripts/order.json [hunk 13]
```diff
@@ -39,8 +51,41 @@
...
             "404": { "type": "unit" }
           }
+        },
+        {
+        "method": "POST",
+          "body": {
+          	"type": "order_form"
+          },
+          "responses": {
+            "200": { "type": "order_form" },
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]:
```suggestion
            "200": { "type": "order" },
```

File: scripts/order.json [hunk 14]
```diff
@@ -39,8 +51,41 @@
...
+            "401": { "type": "unit" },
+            "404": { "type": "unit" },
+            "422": { "type": "error" }
+          }
+        },
+        {
+        "method": "DELETE",
```
Comment (Bhavya-verma-knoldus) [inline scripts/order.json]: Let's reorder this, Let's write `PUT` resource first and then `DELETE`.

**Review by Bhavya-verma-knoldus — APPROVED** [review]

**Pull request merged**